state/factories.go

```diff
@@ -2,13 +2,9 @@ package state
 
 // Factories wraps up all the implementation-specific integration points.
 type Factories interface {
-	NewState() VMWrapper
+	NewStateAndApplier() (VMWrapper, Applier)
```

Comment:
Previously, methods on `Applier` accepted a `VMWrapper`. Now an `Applier` contains a `VMWrapper` and these need to be created at the same time, so this methods was added.